### ISSUE TITLE
Procedure Codes to Visits?

### ISSUE THREAD

**Issue by chrisroederucdenver** [cladteam/CCDA_OMOP_by_Python#177]
Stephanie and Tanner have mentioned an issue that may require more vocabulary mapping. It's either procedure codes or provider codes that somehow work out as misplaced visit codes?

**(new) It's that when you map a procedure code to a visit code for an encounter,  you should create a procedure to more directly, accurately, and specifically record that procedure code. Create the visit as you must, but also create a procedure_occurrence that hangs off that visit for this purpose.**

We should investigate and reconcile the procedure code we find at the encounter level in a CCDA document with any procedures that it contains.

@tannerzhang could you please remind me and clarify this? If you have an example concept and its origin and domain_id, that would help. Thanks.

- [x] clarify the issue (thanks Steph)
- [x] look to see if it occurrs in CCDA data (it's in there)
- [x] identify concepts of values sets involved (that's the VSAC mentioned in #72 and #149)
- [ ] have Tanner map them
- [ ] Reconcile the way these codes appear at the encounter level with what codes may appear in procedures nested within them.

**Comment by chrisroederucdenver** [comment]
from Steph: 
>
the SNOMED Patient encounter procedure codes may be mapped to procedures or a visit domain. Therefore, we cannot rely on these mapping to encode an encounter. The mappings to procedures codes ( HCPCS, CPT4, SNOMED) to OMOP concept that maps to visit domain will be replaced with mappings to procedure Domain. So you will need to create an encounter where these procedure codes can be connected to:
In CCDA, HL7 version 3 base classes- Reference Information Model (RIM) uses ACT code to encode the encounter that occurred.
Here is basic syntax that I can gather about CCDA encounter type documents:
Provides basic syntax for the creation sentences or clinical statements
Act – various actions that can be taken
Mood:
DEF - what should occur
INT - intended to occur
ENV – Acts in event mood describe what has occurred
ENV.CRT – Then event criteria mood is commonly used in CDA to identiry acts that
specify precondition for certain acts (e.g. PRN for pain is a precondition)
In summary, we should only include the encounter that has occurred, i.e. Mood=ENV to capture the encounters.
>

**Comment by chrisroederucdenver** [comment]
See also #181 for being consistent with filtering encounters on "event" mood codes in Xpaths .

**Comment by chrisroederucdenver** [comment]
@stephanieshong  Besides filtering on mood codes, what I get from above is that if you have an encounter that is coded with a procedure code, not only do you have a problem finding or creating the visit_concept_id for that encounter/visit, but you should create a procedure that hangs off of the visit that records the procedure code in a more direct way, as a procedure.
Correct?

**Comment by stephanieshong** [comment]
yes, that is correct. you need that procedure code which hangs off that visit.

**Comment by stephanieshong** [comment]
copied from the other issue, https://github.com/cladteam/CCDA_OMOP_by_Python/issues/72
as these are related: 

Would you be able to provide a list of code/codeSystem (in order) you will be able to use to map the encounter types?

For example for encounter type we can go down the list of coded elements that you can chase after and
COALESCE() the non- null coded values in the clinicalDocumet header that you want to use to encode the encounter type.
And then give us the list of coded value/code System to map to visite_concept_id ( codeSystem will be oid for a known vocabulary Id that we can identify as the well known externally managed terminologySystem or valueSet that we can reference and parse to identify the correct concept id)

related ACTS connected to the CDA document you will have ClincialDocument class that tells you the specific service performed and the encounter in which they took place. This is the enformation we need to assign the encouter type, i.e. visit_concept_id

Important portion you want extracted is the following section:
CDAHeader-> provider, the type of service and the location of service ( often in LOINC classification with type of document, type of service, and the author and location of service provided.
**<tag of the performer or the author role - common service provided by a health care professional may include, care history, physicals, consults, operation or procedures, dishcharge or transfer, progress notes, lab reports, imaging studies, etc.
<tag professional level of the provider - provider level of provider can be physicians, nurses, or residents, etc.
<tag encompassingEncounterlocation act location of the participation class - location of serivce which may include, hospital, home, hospice, long term care facility, Emergency Department or Ambulartory care.

There are complexity that get added with relatedDocuments, but we will worry about that later.
@stephanieshong

**Comment by stephanieshong** [comment]
And from the ClinicalDocument we only need the encounter that has occurred already, i.e. Mood=ENV to capture the encounters that occurred.

**Comment by chrisroederucdenver** [comment]
yep! #181

**Comment by stephanieshong** [comment]
And additionally,  if you can parse the location of the service provided then that value can be used to set the visit_concept_id

**Comment by chrisroederucdenver** [comment]
Deferring for now.  Again, this ticket is about abducting a procedure occurrence from an encounter section or encompassingEncounter header element in addition to the visit_occurrence row we really focussed on in other tickets.
